clients/java/src/main/java/io/zeebe/client/impl/ZeebeClientImpl.java

```diff
@@ -107,7 +110,19 @@ public static ManagedChannel buildChannel(ZeebeClientConfiguration config) {
         NettyChannelBuilder.forAddress(address.getHost(), address.getPort());
 
     if (config.isSecureConnectionEnabled()) {
-      channelBuilder.useTransportSecurity().sslContext(config.getSslContext());
+      final String certificatePath = config.getCaCertificatePath();
+      SslContext sslContext = null;
+
+      if (certificatePath != null && !certificatePath.isEmpty()) {
```

Comment:
Just a thought, but as a user what do we expect? If I say 

```java
ZeebeClient.newClientBuilder()
    .useSecureConnection()
    .caCertificatePath(mySuperCoolVariable)
    .build()
```

But oops `mySuperCoolVariable` is `""` because I'm reading it from some external script that failed. Should it fail? I feel like I would expect it to fail, but that might just be me.

Reply:
Just a thought, why we need to the 'useSecureConnection'? When wr need anyway the certificate? Maybe just certificate is enough? This implies the secure connection right?

Reply:
Having `useSecureConnection` allows us to use TLS without having to specify a custom certificate authority, i.e. just use the system ones, which in most cases should be all you need. So I think that's still useful.

Reply:
Yes we always need the useSecureConnection because we might be using installed certificates. But in the case where we specify a certificate I think there's also a question of consistency with the other places where we specify security configurations, namely the gateway. What would be then the expected behavior in the gateway? If we specify a certificate/private key, TLS is enabled even if we don't explicitly set it? Because for the user that might mean resetting all of the environment variables, instead of just `ZEEBE_GATEWAY_SECURITY_ENABLED` and leaving the certificate and private key path variables still pointing to valid locations.